ejb3/src/main/java/org/jboss/as/ejb3/subsystem/IdentityResourceDefinition.java

```diff
@@ -0,0 +1,204 @@
+            .setAllowNull(true)
+            .setMinSize(1)
+            .setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
+            .setCapabilityReference(SECURITY_DOMAIN_CAPABILITY, IDENTITY_CAPABILITY, false)
+            .build();
+
+    static final IdentityResourceDefinition INSTANCE = new IdentityResourceDefinition();
+
+    private static final AttributeDefinition[] ATTRIBUTES = new AttributeDefinition[] { OUTFLOW_SECURITY_DOMAINS };
+
+    private static List<String> outflowSecurityDomains = Collections.synchronizedList(new ArrayList<>());
+
+    private IdentityResourceDefinition() {
+        this(new Parameters(EJB3SubsystemModel.IDENTITY_PATH, EJB3Extension.getResourceDescriptionResolver(EJB3SubsystemModel.IDENTITY))
+                .setCapabilities(IDENTITY_RUNTIME_CAPABILITY), new AddHandler());
+    }
+
+    private IdentityResourceDefinition(Parameters parameters, AbstractAddStepHandler add) {
+        super(parameters
+                .setAddHandler(add)
+                .setRemoveHandler(new ReloadRequiredRemoveStepHandler(IDENTITY_RUNTIME_CAPABILITY))
+                .setRemoveRestartLevel(OperationEntry.Flag.RESTART_RESOURCE_SERVICES));
+    }
+
+    @Override
+    public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
+        ReloadRequiredWriteAttributeHandler handler = new ReloadRequiredWriteAttributeHandler(ATTRIBUTES);
+        for (AttributeDefinition attribute: ATTRIBUTES) {
+            resourceRegistration.registerReadWriteAttribute(attribute,  null, handler);
+        }
+    }
+
+    private static class AddHandler extends AbstractAddStepHandler {
+
+        private AddHandler() {
+            super(IDENTITY_RUNTIME_CAPABILITY, OUTFLOW_SECURITY_DOMAINS);
+        }
+
+        @Override
+        protected void populateModel(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
+            super.populateModel(context, operation, resource);
+            outflowSecurityDomains = OUTFLOW_SECURITY_DOMAINS.unwrap(context, resource.getModel());
+        }
+
+        @Override
+        protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+            IdentityService identityService = new IdentityService();
+            ServiceBuilder<Function<SecurityIdentity, Set<SecurityIdentity>>> serviceBuilder = context.getServiceTarget()
+                    .addService(IDENTITY_RUNTIME_CAPABILITY.getCapabilityServiceName(), identityService)
+                    .setInitialMode(Mode.ACTIVE);
+            for (String outflowSecurityDomain : outflowSecurityDomains) {
+                serviceBuilder.addDependency(context.getCapabilityServiceName(
+                                buildDynamicCapabilityName(SECURITY_DOMAIN_CAPABILITY, outflowSecurityDomain), SecurityDomain.class),
```

Comment:
This should be 

context.getCapabilityServiceName(SECURITY_DOMAIN_CAPABILITY, outflowSecurityDomain, SecurityDomain.class)

What you have here works, and no critical need to fix it, but if you are changing this class anyway for other things it's more robust to let the OperationContext do things than doing things yourself.